### PR TITLE
[Refactor/#169]:검색한 상태에서 필터 초기화 클릭시 검색창까지 초기화되는 문제 수정

### PR DESCRIPTION
**Pull request by juhee067**
## 📌 작업 내용

검색한 상태에서 필터 초기화 클릭시 검색창까지 초기화되는 문제 수정

<br/>

## 📌 구현 결과 (선택)

구현한 기능이 보이는 스크린샷을 업로드해주세요.

<br/>

## 📌 기타 사항

리뷰어가 특별히 봐주었으면 하는 부분이나 주의사항, 알림사항 등이 있다면
작성해주세요.

<br/>

### CODE REVIEW

**Comment by JIMIN1020** [comment]
<h2>Coverage Report</h2>
<table> <thead> <tr> <th align="center">Status</th> <th align="left">Category</th> <th align="right">Percentage</th> <th align="right">Covered / Total</th> </tr> </thead> <tbody> <tr> <td align="center">🔵</td> <td align="left">Lines</td> <td align="right">7.85%</td> <td align="right">618 / 7867</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Statements</td> <td align="right">7.85%</td> <td align="right">618 / 7867</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Functions</td> <td align="right">15.8%</td> <td align="right">52 / 329</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Branches</td> <td align="right">25.14%</td> <td align="right">89 / 354</td> </tr> </tbody> </table><details><summary>File Coverage</summary><table> <thead> <tr> <th align="left">File</th> <th align="right">Stmts</th> <th align="right">% Branch</th> <th align="right">% Funcs</th> <th align="right">% Lines</th> <th align="left">Uncovered Lines</th> </tr> </thead> <tbody> <tr> <td align="left" colspan="6"><b>Changed Files</b></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts">src/features/Catalog/hooks/useFilter.ts</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts#L1-L2" class="text-red">1-2</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts#L4-L6" class="text-red">4-6</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts#L8-L10" class="text-red">8-10</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts#L12-L19" class="text-red">12-19</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts#L21-L28" class="text-red">21-28</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts#L30-L37" class="text-red">30-37</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts#L39-L40" class="text-red">39-40</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts#L42-L45" class="text-red">42-45</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts#L47-L51" class="text-red">47-51</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts#L53-L55" class="text-red">53-55</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Catalog/hooks/useFilter.ts#L57-L66" class="text-red">57-66</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx">src/features/Join/components/PasswordWithIdForm.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L1-L2" class="text-red">1-2</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L4-L5" class="text-red">4-5</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L7-L9" class="text-red">7-9</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L11-L12" class="text-red">11-12</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L14" class="text-red">14</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L16" class="text-red">16</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L18-L21" class="text-red">18-21</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L23-L24" class="text-red">23-24</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L26-L27" class="text-red">26-27</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L29-L30" class="text-red">29-30</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L32-L33" class="text-red">32-33</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L35" class="text-red">35</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L37-L41" class="text-red">37-41</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L43-L62" class="text-red">43-62</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L64-L80" class="text-red">64-80</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L82-L87" class="text-red">82-87</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L89" class="text-red">89</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L91" class="text-red">91</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L93" class="text-red">93</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L100" class="text-red">100</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L105" class="text-red">105</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L110" class="text-red">110</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L112-L113" class="text-red">112-113</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L116" class="text-red">116</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L120" class="text-red">120</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L124-L125" class="text-red">124-125</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L127" class="text-red">127</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/components/PasswordWithIdForm.tsx#L130-L131" class="text-red">130-131</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/hooks/useSignup.ts">src/features/Join/hooks/useSignup.ts</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/hooks/useSignup.ts#L1-L2" class="text-red">1-2</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/hooks/useSignup.ts#L4-L6" class="text-red">4-6</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/hooks/useSignup.ts#L8-L9" class="text-red">8-9</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/hooks/useSignup.ts#L11-L12" class="text-red">11-12</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/hooks/useSignup.ts#L14-L21" class="text-red">14-21</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/hooks/useSignup.ts#L23-L32" class="text-red">23-32</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/hooks/useSignup.ts#L34-L36" class="text-red">34-36</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/hooks/useSignup.ts#L38-L42" class="text-red">38-42</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a9b581140bbec1a5b7c0a3141e5ec00db7b2d3ce/src/features/Join/hooks/useSignup.ts#L44-L58" class="text-red">44-58</a></td> </tr> </tbody> </table></details>
<em>Generated in workflow <a href=https://github.com/Licruit/Licruit-frontend/actions/runs/10661067097>#256</a></em>

File: src/features/Catalog/hooks/useFilter.ts [hunk 1]
```diff
@@ -37,9 +37,21 @@ export const useFilter = () => {
...
+    searchParams.delete('minAlcohol');
+    searchParams.delete('maxAlcohol');
+    searchParams.delete('sort');
+    searchParams.delete('page');
+
+    if (searchQuery) {
+      navigate(`/catalog?search=${searchQuery}`);
+    } else {
+      navigate(`/catalog`);
+    }
```
Comment (JIMIN1020) [inline src/features/Catalog/hooks/useFilter.ts:51]: PATH 상수 사용하기!

**Review by JIMIN1020 — APPROVED** [review]
마지막까지 수정하느라 고생이 많습니다요

**Review by SebellKo — APPROVED** [review]
주히님 고생했어요 ~

**Pull request merged**